### PR TITLE
VLAN dependency handling for Dynamic Port Breakout

### PR DESCRIPTION
**Pull request by vasant17**
Set port::VLAN_DEP bit when port is added to first VLAN
    Clear port::VLAN_DEP bit when port is removed from all VLANs
During port removal check for the dependency bit map.

Also added, pretty printing of dependency list

**What I did**
1. Code changes in portsorch to handle VLAN dependency on port.
    -Set port::VLAN_DEP bit when port is added to first VLAN
    -Clear port::VLAN_DEP bit when port is removed from all VLANs . 
    -During port removal check for the dependency bit map.

2. Also added pretty printing of dependency list
3. Refactored DPB test infra code.
4. Added test cases for VLAN dependency testing

**Why I did it**
To support DPB feature
**How I verified it**
Ran all VS test cases, and also newly added VLAN DPB test cases.

**Details if related**
======================================================================= test session starts ========================================================================
platform linux2 -- Python 2.7.15+, pytest-3.3.0, py-1.8.0, pluggy-0.6.0 -- /usr/bin/python                                                                          
cachedir: .cache                                                                                                                                                    
rootdir: /home/vapatil/workspace/XU_DPB/sonic-buildimage/src/sonic-swss/tests, inifile:
collected 182 items                                                                                                                                                

test_acl.py::TestAcl::test_AclTableCreation PASSED                                                                                                           [  0%]
test_acl.py::TestAcl::test_AclRuleL4SrcPort PASSED                                                                                                           [  1%]
test_acl.py::TestAcl::test_AclRuleInOutPorts PASSED                                                                                                          [  1%]
test_acl.py::TestAcl::test_AclTableDeletion PASSED                                                                                                           [  2%]
test_acl.py::TestAcl::test_V6AclTableCreation PASSED                                                                                                         [  2%]
test_acl.py::TestAcl::test_V6AclRuleIPv6Any PASSED                                                                                                           [  3%]
test_acl.py::TestAcl::test_V6AclRuleIPv6AnyDrop PASSED                                                                                                       [  3%]
test_acl.py::TestAcl::test_V6AclRuleIpProtocol PASSED                                                                                                        [  4%]
test_acl.py::TestAcl::test_V6AclRuleSrcIPv6 PASSED                                                                                                           [  4%]
test_acl.py::TestAcl::test_V6AclRuleDstIPv6 PASSED                                                                                                           [  5%]
test_acl.py::TestAcl::test_V6AclRuleL4SrcPort PASSED                                                                                                         [  6%]
test_acl.py::TestAcl::test_V6AclRuleL4DstPort PASSED                                                                                                         [  6%]
test_acl.py::TestAcl::test_V6AclRuleTCPFlags PASSED                                                                                                          [  7%]
test_acl.py::TestAcl::test_V6AclRuleL4SrcPortRange PASSED                                                                                                    [  7%]
test_acl.py::TestAcl::test_V6AclRuleL4DstPortRange PASSED                                                                                                    [  8%]
test_acl.py::TestAcl::test_V6AclTableDeletion PASSED                                                                                                         [  8%]
test_acl.py::TestAcl::test_InsertAclRuleBetweenPriorities PASSED                                                                                             [  9%]
test_acl.py::TestAcl::test_RulesWithDiffMaskLengths PASSED                                                                                                   [  9%]
test_acl.py::TestAcl::test_AclRuleIcmp PASSED                                                                                                                [ 10%]
test_acl.py::TestAcl::test_AclRuleIcmpV6 PASSED                                                                                                              [ 10%]
test_acl.py::TestAclRuleValidation::test_AclActionValidation PASSED                                                                                          [ 11%]
test_acl_ctrl.py::TestPortChannelAcl::test_AclCtrl PASSED                                                                                                    [ 12%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclTableCreation PASSED                                                                             [ 12%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclRuleL4SrcPortRange PASSED                                                                        [ 13%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclRuleL4DstPortRange PASSED                                                                        [ 13%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclRuleL2EthType PASSED                                                                             [ 14%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclRuleTunnelVNI PASSED                                                                             [ 14%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclRuleTC PASSED                                                                                    [ 15%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclInnerIPProtocol PASSED                                                                           [ 15%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclInnerEthType PASSED                                                                              [ 16%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclInnerL4SrcPort PASSED                                                                            [ 17%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclInnerL4DstPort PASSED                                                                            [ 17%]
test_acl_egress_table.py::TestEgressAclTable::test_EgressAclTableDeletion PASSED                                                                             [ 18%]
test_acl_portchannel.py::TestPortChannelAcl::test_PortChannelAfterAcl PASSED                                                                                 [ 18%]
test_acl_portchannel.py::TestPortChannelAcl::test_PortChannelBeforeAcl PASSED                                                                                [ 19%]
test_acl_portchannel.py::TestPortChannelAcl::test_AclOnPortChannelMember PASSED                                                                              [ 19%]
test_admin_status.py::TestAdminStatus::test_PortChannelMemberAdminStatus PASSED                                                                              [ 20%]
test_crm.py::TestCrm::test_CrmFdbEntry PASSED                                                                                                                [ 20%]
test_crm.py::TestCrm::test_CrmIpv4Route PASSED                                                                                                               [ 21%]
test_crm.py::TestCrm::test_CrmIpv6Route PASSED                                                                                                               [ 21%]
test_crm.py::TestCrm::test_CrmIpv4Nexthop PASSED                                                                                                             [ 22%]
test_crm.py::TestCrm::test_CrmIpv6Nexthop PASSED                                                                                                             [ 23%]
test_crm.py::TestCrm::test_CrmIpv4Neighbor PASSED                                                                                                            [ 23%]
test_crm.py::TestCrm::test_CrmIpv6Neighbor PASSED                                                                                                            [ 24%]
test_crm.py::TestCrm::test_CrmNexthopGroup PASSED                                                                                                            [ 24%]
test_crm.py::TestCrm::test_CrmNexthopGroupMember PASSED                                                                                                      [ 25%]
test_crm.py::TestCrm::test_CrmAcl PASSED                                                                                                                     [ 25%]
test_crm.py::TestCrm::test_CrmAclGroup PASSED                                                                                                                [ 26%]
test_crm.py::TestCrm::test_Configure PASSED                                                                                                                  [ 26%]
test_crm.py::TestCrm::test_Configure_ipv4_route PASSED                                                                                                       [ 27%]
test_crm.py::TestCrm::test_Configure_ipv6_route PASSED                                                                                                       [ 28%]
test_crm.py::TestCrm::test_Configure_ipv4_nexthop PASSED                                                                                                     [ 28%]
test_crm.py::TestCrm::test_Configure_ipv6_nexthop PASSED                                                                                                     [ 29%]
test_crm.py::TestCrm::test_Configure_ipv4_neighbor PASSED                                                                                                    [ 29%]
test_crm.py::TestCrm::test_Configure_ipv6_neighbor PASSED                                                                                                    [ 30%]
test_crm.py::TestCrm::test_Configure_group_member PASSED                                                                                                     [ 30%]
test_crm.py::TestCrm::test_Configure_group_object PASSED                                                                                                     [ 31%]
test_crm.py::TestCrm::test_Configure_acl_table PASSED                                                                                                        [ 31%]
test_crm.py::TestCrm::test_Configure_acl_group PASSED                                                                                                        [ 32%]
test_crm.py::TestCrm::test_Configure_acl_group_entry PASSED                                                                                                  [ 32%]
test_crm.py::TestCrm::test_Configure_acl_group_counter PASSED                                                                                                [ 33%]
test_crm.py::TestCrm::test_Configure_fdb PASSED                                                                                                              [ 34%]
test_dirbcast.py::TestDirectedBroadcast::test_DirectedBroadcast PASSED                                                                                       [ 34%]
test_dtel.py::TestDtel::test_DtelGlobalAttribs PASSED                                                                                                        [ 35%]
test_dtel.py::TestDtel::test_DtelReportSessionAttribs PASSED                                                                                                 [ 35%]
test_dtel.py::TestDtel::test_DtelINTSessionAttribs PASSED                                                                                                    [ 36%]
test_dtel.py::TestDtel::test_DtelQueueReportAttribs PASSED                                                                                                   [ 36%]
test_dtel.py::TestDtel::test_DtelEventAttribs PASSED                                                                                                         [ 37%]
test_fdb.py::TestFdb::test_FdbWarmRestartNotifications PASSED                                                                                                [ 37%]
test_fdb.py::TestFdb::test_FdbAddedAfterMemberCreated PASSED                                                                                                 [ 38%]
test_fdb_update.py::test_FDBAddedAndUpdated PASSED                                                                                                           [ 39%]
test_fdb_update.py::test_FDBLearnedAndUpdated PASSED                                                                                                         [ 39%]
test_interface.py::TestRouterInterface::test_InterfaceAddRemoveIpv6Address PASSED                                                                            [ 40%]
test_interface.py::TestRouterInterface::test_InterfaceAddRemoveIpv4Address PASSED                                                                            [ 40%]
test_interface.py::TestRouterInterface::test_InterfaceSetMtu PASSED                                                                                          [ 41%]
test_interface.py::TestLagRouterInterfaceIpv4::test_InterfaceAddRemoveIpv4Address PASSED                                                                     [ 41%]
test_interface.py::TestLagRouterInterfaceIpv4::test_InterfaceSetMtu PASSED                                                                                   [ 42%]
test_interface.py::TestLoopbackRouterInterface::test_InterfacesCreateRemove PASSED                                                                           [ 42%]
test_mirror.py::TestMirror::test_MirrorAddRemove PASSED                                                                                                      [ 43%]
test_mirror.py::TestMirror::test_MirrorToVlanAddRemove PASSED                                                                                                [ 43%]
test_mirror.py::TestMirror::test_MirrorToLagAddRemove PASSED                                                                                                 [ 44%]
test_mirror.py::TestMirror::test_MirrorDestMoveVlan PASSED                                                                                                   [ 45%]
test_mirror.py::TestMirror::test_MirrorDestMoveLag PASSED                                                                                                    [ 45%]
test_mirror.py::TestMirror::test_AclBindMirrorPerStage PASSED                                                                                                [ 46%]
test_mirror.py::TestMirror::test_AclBindMirror PASSED                                                                                                        [ 46%]
test_mirror_ipv6_combined.py::TestMirror::test_AclBindMirrorV6Combined FAILED                                                                                [ 47%]
test_mirror_ipv6_combined.py::TestMirror::test_AclBindMirrorV6Reorder1 FAILED                                                                                [ 47%]
test_mirror_ipv6_combined.py::TestMirror::test_AclBindMirrorV6Reorder2 FAILED                                                                                [ 48%]
test_mirror_ipv6_combined.py::TestMirror::test_AclBindMirrorV6WrongConfig FAILED                                                                             [ 48%]
test_mirror_ipv6_separate.py::TestMirror::test_AclBindMirrorSeparated FAILED                                                                                 [ 49%]
test_mirror_ipv6_separate.py::TestMirror::test_AclBindMirrorV6Reorder1 FAILED                                                                                [ 50%]
test_mirror_ipv6_separate.py::TestMirror::test_AclBindMirrorV6Reorder2 FAILED                                                                                [ 50%]
test_mirror_ipv6_separate.py::TestMirror::test_AclBindMirrorV6WrongConfig FAILED                                                                             [ 51%]
test_mirror_policer.py::TestMirror::test_MirrorPolicer PASSED                                                                                                [ 51%]
test_mirror_policer.py::TestMirror::test_MirrorPolicerWithAcl PASSED                                                                                         [ 52%]
test_nhg.py::TestNextHopGroup::test_route_nhg PASSED                                                                                                         [ 52%]
test_pfc.py::TestPfc::test_PfcAsymmetric PASSED                                                                                                              [ 53%]
test_policer.py::TestPolicer::test_PolicerBasic PASSED                                                                                                       [ 53%]
test_port.py::TestPort::test_PortMtu PASSED                                                                                                                  [ 54%]
test_port.py::TestPort::test_PortNotification PASSED                                                                                                         [ 54%]
test_port.py::TestPort::test_PortFec PASSED                                                                                                                  [ 55%]
test_port.py::TestPort::test_PortPreemp PASSED                                                                                                               [ 56%]
test_port.py::TestPort::test_PortIdriver PASSED                                                                                                              [ 56%]
test_port.py::TestPort::test_PortIpredriver PASSED                                                                                                           [ 57%]
test_port_an.py::TestPortAutoNeg::test_PortAutoNegCold PASSED                                                                                                [ 57%]
test_port_an.py::TestPortAutoNeg::test_PortAutoNegWarm PASSED                                                                                                [ 58%]
test_port_buffer_rel.py::TestPortBuffer::test_PortsAreUpAfterBuffers PASSED                                                                                  [ 58%]
test_port_config.py::TestPortConfig::test_port_hw_lane PASSED                                                                                                [ 59%]
test_port_config.py::TestPortConfig::test_port_breakout PASSED                                                                                               [ 59%]
test_port_dpb.py::TestPortDPB::test_port_breakout_one PASSED                                                                                                 [ 60%]
test_port_dpb.py::TestPortDPB::test_port_breakout_multiple vapatil
PASSED                                                                                            [ 60%]
test_port_dpb.py::TestPortDPB::test_port_breakout_all SKIPPED                                                                                                [ 61%]
test_port_dpb_acl.py::TestPortDPBAcl::test_acl_table_empty_port_list PASSED                                                                                  [ 62%]
test_port_dpb_acl.py::TestPortDPBAcl::test_one_port_two_acl_tables PASSED                                                                                    [ 62%]
test_port_dpb_acl.py::TestPortDPBAcl::test_one_acl_table_many_ports PASSED                                                                                   [ 63%]
test_port_dpb_acl.py::TestPortDPBAcl::test_one_port_many_acl_tables SKIPPED                                                                                  [ 63%]
test_port_dpb_vlan.py::TestPortDPBVlan::test_dependency FAILED                                                                                               [ 64%]
test_port_dpb_vlan.py::TestPortDPBVlan::test_one_port_one_vlan FAILED                                                                                        [ 64%]
test_portchannel.py::TestPortchannel::test_Portchannel PASSED                                                                                                [ 65%]
test_portchannel.py::TestPortchannel::test_Portchannel_oper_down PASSED                                                                                      [ 65%]
test_qos_map.py::TestDot1p::test_dot1p_cfg PASSED                                                                                                            [ 66%]
test_qos_map.py::TestDot1p::test_port_dot1p PASSED                                                                                                           [ 67%]
test_route.py::TestRoute::test_RouteAdd PASSED                                                                                                               [ 67%]
test_setro.py::TestSetRo::test_SetReadOnlyAttribute PASSED                                                                                                   [ 68%]
test_sflow.py::TestSflow::test_SflowDisble PASSED                                                                                                            [ 68%]
test_sflow.py::TestSflow::test_InterfaceSet PASSED                                                                                                           [ 69%]
test_sflow.py::TestSflow::test_ConfigDel PASSED                                                                                                              [ 69%]
test_speed.py::TestSpeedSet::test_SpeedAndBufferSet PASSED                                                                                                   [ 70%]
test_switch.py::TestSwitch::test_switch_attribute PASSED                                                                                                     [ 70%]
test_tunnel.py::TestDecapTunnel::test_TunnelDecap_v4 PASSED                                                                                                  [ 71%]
test_tunnel.py::TestDecapTunnel::test_TunnelDecap_v6 PASSED                                                                                                  [ 71%]
test_tunnel.py::TestSymmetricTunnel::test_TunnelSymmetric_v4 PASSED                                                                                          [ 72%]
test_tunnel.py::TestSymmetricTunnel::test_TunnelSymmetric_v6 PASSED                                                                                          [ 73%]
test_vlan.py::TestVlan::test_VlanAddRemove PASSED                                                                                                            [ 73%]
test_vlan.py::TestVlan::test_MultipleVlan PASSED                                                                                                             [ 74%]
test_vlan.py::TestVlan::test_VlanIncrementalConfig PASSED                                                                                                    [ 74%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectKeyPrefix[test_input0-0] PASSED                                                                             [ 75%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectKeyPrefix[test_input1-0] PASSED                                                                             [ 75%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectKeyPrefix[test_input2-0] PASSED                                                                             [ 76%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectKeyPrefix[test_input3-1] PASSED                                                                             [ 76%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectValueType[test_input0-0] PASSED                                                                             [ 77%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectValueType[test_input1-0] PASSED                                                                             [ 78%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectValueType[test_input2-0] PASSED                                                                             [ 78%]
test_vlan.py::TestVlan::test_AddVlanWithIncorrectValueType[test_input3-1] PASSED                                                                             [ 79%]
test_vlan.py::TestVlan::test_AddPortChannelToVlan PASSED                                                                                                     [ 79%]
test_vlan.py::TestVlan::test_AddVlanMemberWithNonExistVlan PASSED                                                                                            [ 80%]
test_vlan.py::TestVlan::test_RemoveNonexistentVlan PASSED                                                                                                    [ 80%]
test_vlan.py::TestVlan::test_VlanMemberTaggingMode[test_input0-expected0] PASSED                                                                             [ 81%]
test_vlan.py::TestVlan::test_VlanMemberTaggingMode[test_input1-expected1] PASSED                                                                             [ 81%]
test_vlan.py::TestVlan::test_VlanMemberTaggingMode[test_input2-expected2] PASSED                                                                             [ 82%]
test_vlan.py::TestVlan::test_VlanMemberTaggingMode[test_input3-expected3] PASSED                                                                             [ 82%]
test_vlan.py::TestVlan::test_VlanMemberTaggingMode[test_input4-expected4] PASSED                                                                             [ 83%]
test_vlan.py::TestVlan::test_AddMaxVlan SKIPPED                                                                                                              [ 84%]
test_vlan.py::TestVlan::test_RemoveVlanWithRouterInterface PASSED                                                                                            [ 84%]
test_vlan.py::TestVlan::test_VlanDbData PASSED                                                                                                               [ 85%]
test_vlan.py::TestVlan::test_VlanMemberDbData[test_input0-expected0] PASSED                                                                                  [ 85%]
test_vlan.py::TestVlan::test_VlanMemberDbData[test_input1-expected1] PASSED                                                                                  [ 86%]
test_vlan.py::TestVlan::test_VlanMemberDbData[test_input2-expected2] PASSED                                                                                  [ 86%]
test_vnet.py::TestVnetOrch::test_vnet_orch_1 PASSED                                                                                                          [ 87%]
test_vnet.py::TestVnetOrch::test_vnet_orch_2 PASSED                                                                                                          [ 87%]
test_vnet.py::TestVnetOrch::test_vnet_orch_3 PASSED                                                                                                          [ 88%]
test_vnet.py::TestVnetOrch::test_vnet_orch_4 SKIPPED                                                                                                         [ 89%]
test_vnet.py::TestVnetOrch::test_vnet_orch_5 PASSED                                                                                                          [ 89%]
test_vnet_bitmap.py::TestVnetBitmapOrch::test_vnet_orch_1 <- test_vnet.py FAILED                                                                             [ 90%]
test_vnet_bitmap.py::TestVnetBitmapOrch::test_vnet_orch_2 <- test_vnet.py FAILED                                                                             [ 90%]
test_vnet_bitmap.py::TestVnetBitmapOrch::test_vnet_orch_3 <- test_vnet.py FAILED                                                                             [ 91%]
test_vnet_bitmap.py::TestVnetBitmapOrch::test_vnet_orch_4 <- test_vnet.py SKIPPED                                                                            [ 91%]
test_vnet_bitmap.py::TestVnetBitmapOrch::test_vnet_orch_5 <- test_vnet.py FAILED                                                                             [ 92%]
test_vrf.py::TestVrf::test_VRFOrch_Comprehensive PASSED                                                                                                      [ 92%]
test_vrf.py::TestVrf::test_VRFOrch PASSED                                                                                                                    [ 93%]
test_vrf.py::TestVrf::test_VRFOrch_Update PASSED                                                                                                             [ 93%]
test_vxlan_tunnel.py::TestVxlan::test_vxlan_term_orch PASSED                                                                                                 [ 94%]
test_warm_reboot.py::TestWarmReboot::test_PortSyncdWarmRestart PASSED                                                                                        [ 95%]
test_warm_reboot.py::TestWarmReboot::test_VlanMgrdWarmRestart PASSED                                                                                         [ 95%]
test_warm_reboot.py::TestWarmReboot::test_swss_neighbor_syncup PASSED                                                                                        [ 96%]
test_warm_reboot.py::TestWarmReboot::test_OrchagentWarmRestartReadyCheck PASSED                                                                              [ 96%]
test_warm_reboot.py::TestWarmReboot::test_swss_port_state_syncup PASSED                                                                                      [ 97%]
test_warm_reboot.py::TestWarmReboot::test_routing_WarmRestart PASSED                                                                                         [ 97%]
test_warm_reboot.py::TestWarmReboot::test_system_warmreboot_neighbor_syncup PASSED                                                                           [ 98%]
test_watermark.py::TestWatermark::test_telemetry_period PASSED                                                                                               [ 98%]
test_watermark.py::TestWatermark::test_lua_plugins SKIPPED                                                                                                   [ 99%]
test_watermark.py::TestWatermark::test_clear SKIPPED                                                                                                         [100%]

14 failed, 161 passed, 7 skipped in 6888.49 seconds
Failed test cases:
8 -> Mirror test cases
4 -> Vnet bitmap test cases
2-> Newly added VLAN DPB test cases. Issue was in script, fixed and re-ran the test cases. They were success.
=============================================================================================== test session starts ================================================================================================
platform linux2 -- Python 2.7.15+, pytest-3.3.0, py-1.8.0, pluggy-0.6.0 -- /usr/bin/python
cachedir: .cache
rootdir: /home/vapatil/workspace/XU_DPB/sonic-buildimage/src/sonic-swss/tests, inifile:
collected 3 items                                                                                                                                                                                                  

test_port_dpb_vlan.py::TestPortDPBVlan::test_dependency remove extra link dummy
PASSED                                                                                                                                               [ 33%]
test_port_dpb_vlan.py::TestPortDPBVlan::test_one_port_one_vlan PASSED                                                                                                                                        [ 66%]
test_port_dpb_vlan.py::TestPortDPBVlan::test_one_port_multiple_vlan SKIPPED                                                                                                                                  [100%]

====================================================================================== 2 passed, 1 skipped in 121.09 seconds =======================================================================================

Updated Test results for VLAN DPB test cases:
vapatil@server09:~/workspace/XU_DPB/sonic-buildimage/src/sonic-swss/tests$ sudo pytest -s -v --dvsname=vs-vp --pdb test_port_dpb_vlan.py
=============================================================================================== test session starts ================================================================================================
platform linux2 -- Python 2.7.15+, pytest-3.3.0, py-1.8.0, pluggy-0.6.0 -- /usr/bin/python
cachedir: .cache
rootdir: /home/vapatil/workspace/XU_DPB/sonic-buildimage/src/sonic-swss/tests, inifile:
collected 3 items                                                                                                                                                                                                  

test_port_dpb_vlan.py::TestPortDPBVlan::test_dependency remove extra link dummy
Exception AssertionError: AssertionError(u"assert 0 > 0\n +  where 0 = int('0')\n +    where '0' = <built-in method group of _sre.SRE_Match object at 0x7f22476d1dc8>(1)\n +      where <built-in method group of _sre.SRE_Match object at 0x7f22476d1dc8> = <_sre.SRE_Match object at 0x7f22476d1dc8>.group",) in <bound method ApplDbValidator.__del__ of <conftest.ApplDbValidator object at 0x7f22477d1250>> ignored
PASSED                                                                                                                                               [ 33%]
test_port_dpb_vlan.py::TestPortDPBVlan::test_one_port_one_vlan PASSED                                                                                                                                        [ 66%]
test_port_dpb_vlan.py::TestPortDPBVlan::test_one_port_multiple_vlan PASSED                                                                                                                                   [100%]Exception AssertionError: AssertionError(u"assert 0 > 0\n +  where 0 = int('0')\n +    where '0' = <built-in method group of _sre.SRE_Match object at 0x7f22476b2378>(1)\n +      where <built-in method group of _sre.SRE_Match object at 0x7f22476b2378> = <_sre.SRE_Match object at 0x7f22476b2378>.group",) in <bound method ApplDbValidator.__del__ of <conftest.ApplDbValidator object at 0x7f224769f390>> ignored

============================================================================================ 3 passed in 156.82 seconds ============================================================================================
vapatil@server09:~/workspace/XU_DPB/sonic-buildimage/src/sonic-swss/tests$

### CODE REVIEW

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -2348,9 +2349,14 @@ void PortsOrch::doVlanMemberTask(Consumer &consumer)
...
 
             if (addBridgePort(port) && addVlanMember(vlan, port, tagging_mode))
+            {
+                m_portList[port.m_alias].set_dependency(Port::VLAN_DEP);
                 it = consumer.m_toSync.erase(it);
+            }
             else
-                it++;
+            {
+                it++; //continue?
```
Comment (zhenggen-xu) [inline orchagent/portsorch.cpp]: it will continue automatically, remove the comment.

File: tests/test_port_dpb_vlan.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        #print "Ethernet0 removed from VLAN101 and its still present in ASIC DB"
+
+        dvs.remove_vlan_member("102", "Ethernet0")
+        assert(p.exists_in_asic_db() == False)
+        #print "Ethernet0 removed from VLAN101 and also from ASIC DB"
+
+        dpb.create_child_ports(dvs, p, 4)
+        #print "1X40G ---> 4x10G verified"
+
+        # Breakin
```
Comment (zhenggen-xu) [inline tests/test_port_dpb_vlan.py:154]: What are we trying to do here with Breakin?

Comment (vasant17) [reply tests/test_port_dpb_vlan.py:154]: Line 149, did breakout of Ethernet0 from 1X40G --> 4X10G. Now, we need to move back to original state. 4X10G --> 1X40G. Every test case I am making sure that I go back to original state.

File: tests/test_port_dpb_vlan.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        dvs.remove_vlan_member("100", "Ethernet0")
+        assert(p.exists_in_asic_db() == True)
+        #print "Ethernet0 removed from VLAN100 and its still present in ASIC DB"
+
+        dvs.remove_vlan_member("101", "Ethernet0")
+        assert(p.exists_in_asic_db() == True)
+        #print "Ethernet0 removed from VLAN101 and its still present in ASIC DB"
+
+        dvs.remove_vlan_member("102", "Ethernet0")
+        assert(p.exists_in_asic_db() == False)
```
Comment (zhenggen-xu) [inline tests/test_port_dpb_vlan.py:148]: sleep 1  here?

File: tests/test_port_dpb_vlan.py (hunk 2 above)
Comment (zhenggen-xu) [reply tests/test_port_dpb_vlan.py:154]: I just had doubt that the below code was not completed or verified.   If you see it is good, please comment out "@pytest.mark.skip()"

**Review by zhenggen-xu — APPROVED** [review]

**Pull request merged**